sechub-doc/src/docs/asciidoc/documents/shared/configuration/executor_config.adoc

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+[[section-executor-configuration]]
+=== Executor configuration
+{sechub} uses `product executors` to call security products by dedicated `product adapters`.
+Each used security product has at least one executor configuration.
+
+Inside such a configuration, we can define properties and parameters which are used when it comes
+to {sechub} job execution. The data is evaluated and used by product executors at runtime and no server restart is necessary.
+
+So for example you can easily change a base URL of a product location and it will automatically be
+used by next job execution.
+
+[WARNING]
+====
+`NESSUS` and `NETSPARKER` product executors does currently not use
+all properties and parameters but needs still some dedicated ENV entries! But you have
+to define a configuration - and a profile - for those products as well!
+
+====
+
+==== Data structure
+
+===== Name
+Just the name of the executor configuration. Can be changed
+
+===== Enabled
+Only when the executor configuration is enabled, it will be used (of courseit has also
+be added to an enabled profile and this to a project).
+
+===== UUID
+Unmodifiaable unique ID of an executor configuration. Cannot be changed. 
+
+===== Product identifier
+The product identifier for the product executor to use.
+Examples are: 
+
+- `CHECKMARX`
+- `NESSUS`
+- `NETSPARKER`
+- `PDS_CODESCAN`
+- `PDS_WEBSCAN`
+- `PDS_INFRCAN`
+
+===== Executor version
+At the moment all product executors exists only in version `1` and as long as there is no
+breaking change the exisitng product executor will be extended and the existing version will be kept.
+
+When there are breaking changes - e.g. the communication between {sechub} and a new version of the security
+product becomes completely different, or parameters have a complete other meaining, an additional
+product executor with another version will be available.
+
+===== Product base URL
+Here we define the base URL to the security product including port number.
+
+As an example: `https://checkmarx.example.org:6011`
+
+===== Product credentials 
+We need dedicated credentials to communicate/use a security product.
+
+====== User
+Just the user name for product. You can either define here the real user 
+name or use `env:ENVIRONMENT_VARIABLENAME` instead. The environment name
+must be set at SecHub server side with origin user name so the server will
+store `env:ENVIRONMENT_VARIABLENAME` as configuration in database, but will
+use the content of environment variable `ENVIRONMENT_VARIABLENAME` at 
+job execution runtime.
```

Comment:
How is it possible to have the `ENVIRONMENT_VARIABLENAME` available during job execution? Previously you mentioned SecHub server side. Are those variables transferred from SecHub to the PDS? Are we talking about executing a job on the PDS? Please explain in more detail what job execution runtime means in this context. It might be obvious for you, but it is not for me.

Reply:
I rewrote the complete part, added also an example: 

```asciidoc
*Example for an environment variable reference:* 

Let's assume a have a running PDS server which accepts _"exampleToken"_ as API token for 
technical user. 

You would add
---- 
export PDS_CODESCAN1_APITOKEN=exampleToken
----
at the beginning of your start script for {sechub} server.

In your executor configuration you would set the API token field to `env:PDS_CODESCAN1_APITOKEN`.
When a {sechub} job has beend started, {sechub} server will use `exampleToken` as
API token for communication with this PDS server.
```

Reply:
done